### DISCUSSION TITLE
CloudFlare Transform Rule interaction with <ClientOnly>

### DISCUSSION THREAD

**Discussion by ms264556** [vuejs/vitepress#3892] [Q&A]
I have a Cloudflare Transform Rule so that some visitors to page /XXX actually see page /YYY. These are not redirects: the URL still shows /XXX.

My problem is that \<ClientOnly> content uses the URL to decide when to apply, so the client-side content from page /XXX clobbers the content from /YYY (which has no client-side content of its own).

Is there any workaround you can recommend to avoid this behaviour? Maybe by teleporting?

**Comment by brc-dd** [comment]
Not sure but depending on your use case, you might be able to use [vitepress rewrites](https://vitepress.dev/guide/routing#route-rewrites) instead of transform rules
